scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/CertificateMessage.java

```diff
@@ -286,7 +287,8 @@ public CertPath getCertificateChain() {
 	 * 
 	 * @throws HandshakeException if any of the checks fails
 	 */
-	public void verifyCertificate(X509Certificate[] trustedCertificates) throws HandshakeException {
+	public void verifyCertificate(X509Certificate[] trustedCertificates, 
```

Comment:
OK, I see the "collection/array" solution was also chosen for x.509.
Maybe, we add a "common" trust interface with two different methods for x509 and rpk.

Reply:
I didn't think to this so much but intuitively I will go for separate interface for RPK and x509.
With maybe a default class which implement both.

Reply:
FMPOV: there are two places, where the one or two interfaces occurs:
- the implementation, but if this implements both, it seems quite similar
- the  verifyCertificate(X509validator, RPKvalidator) then be called with the same "multi validator" for both arguments.

for the first, I agree, but for the second, where we the pass in the same twice, I'm not sure, if the separate interface is more intuitive.

Reply:
For two interfaces:
Maybe we better move the verifyCertificate from CertificateMessage to the Handshaker, which should then use the right interface to validate the certPath or the RPK.

Reply:
Thinking a bit more about that and as we have a specific interface for psk, it make sense to me to have 2 others specific interfaces for RPK and x509.

Finally, a default class which implements both interface make no so much sense.
I suppose the default one for RPK will be "trust-all" and no default one for certificates.
Both could have a static implementations which take list of arrays/list of trusted elements (rpk/certificate).
Builder.setTrustStore() could be a shortcut to create this static store for certificate.

Reply:
As I wrote: going for two interfaces, I would move the verifyCertificate from CertificateMessage to the Handshaker.

Reply:
How do you suggest to transfer the necessary parameters from CertificateMessage?
Currently verifyCertificate() depends on:
1.  CertificateMessage.certPath
2.  AbstractMessage.getPeer()
3. CertificateMessage. rawPublicKeyBytes
4. CertificateMessage.getPublicKey()

Reply:
1. `CertificateMessage.getCertificateChain()` returns the `certPath`, if available.
2. the peer seems only be required for alert message. The Handshaker already creates such alert messages using the peer from the session, see `Handshaker.wrapMessage`
3. if `CertificateMessage.getCertificateChain()` returns `null`, `CertificateMessage.getPublicKey()` will return also `null`, if `rawPublicKeyBytes` is not set. So only 
4. `CertificateMessage.getPublicKey()` is required.

FMPOV, if we use two interfaces for validating the different certificate types, it seems for me to be not that natural to provide both to one method. And the stuff inside that `CertificateMessage.verifyCertificate()` doesn't depend on too much internals. Therefore I would prefer to move it into the Handshaker. But that's just my opinion.

Reply:
Ok I've tried that option out and codewise it works. However CertificateMessageTest is seriously broken by that change and needs to be reworked in concert with HandshakerTest